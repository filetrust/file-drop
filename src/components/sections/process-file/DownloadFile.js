import React from 'react';
import { trackPromise } from 'react-promise-tracker';
import { engineApi } from '../../../api';

class DownloadFile extends React.Component {
	getProtectedFile = () => {
		trackPromise(
      engineApi.protectFile(this.props.file)
			.then(blob => {
				let url = window.URL.createObjectURL(blob);
				let a = document.createElement('a');
				a.href = url;
				a.download = this.props.file.name;
				a.click();
			})
			.catch((error) => {
					console.log(error)
			}))
	}

  render() {
    if (!this.props.hasIssues) {
      return  <button className="button button-filled button-icon left" onClick={this.getProtectedFile}>Download Protected File</button>
	}
	return null

    }
}

export default DownloadFile;
