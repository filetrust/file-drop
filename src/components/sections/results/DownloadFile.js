import React from 'react';
import { trackPromise } from 'react-promise-tracker';
import { engineApi } from '../../../api';
import { Button } from '../../widgets';

class DownloadFile extends React.Component {
    getProtectedFile = () => {
        trackPromise(
            engineApi.protectFile(this.props.file)
            .then(blob => {
                let url = window.URL.createObjectURL(blob);
                let a = document.createElement('a');
                a.href = url;
                a.download = this.props.file.name;
                a.click();
            })
            .catch((error) => {
                // debugger;
                console.log(error.message)
            }))
    }

    render() {
        if ( !this.props.hasIssues ) {
            return <Button inverse onClick={this.getProtectedFile}>Download Protected File</Button>
        }
        return null
    }
}

export default DownloadFile;
