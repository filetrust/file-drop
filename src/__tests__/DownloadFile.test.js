import React from "react";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import DownloadFile from "../components/DownloadFile";

Enzyme.configure({adapter: new Adapter()});

describe("DownloadFile", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<DownloadFile />);
    expect(wrapper.exists()).toBe(true);
  });

  it("matches snapshot", () => {
    const wrapper = shallow(<DownloadFile />);
    expect(wrapper).toMatchSnapshot();
  });

  it("shows button when file has been protected", () => {
    const wrapper = shallow(<DownloadFile hasIssues={false} />);
    expect(wrapper.contains(<button>Download Protected File</button>));
  });
});
