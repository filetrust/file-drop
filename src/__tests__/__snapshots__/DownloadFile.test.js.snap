// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadFile matches snapshot 1`] = `ShallowWrapper {}`;
